synapse/datamodel.py

```diff
@@ -904,14 +911,18 @@ def _addFormIface(self, form, name):
             mesg = f'Form {form.name} depends on non-existant interface: {name}'
             raise s_exc.NoSuchName(mesg=mesg)
 
+        if iface.get('deprecated'):
+            mesg = f'Form {form.name} depends on deprecated interface {name} which will be removed in 3.0.0'
```

Comment:
Seems like this could get spammy in the logs. Should we iterate over all the loaded interfaces and print a count of deprecated interaces similar to `synapse.cortex.Cortex._warnDeprLocks()`?

Reply:
Actually, `_warnDeprLocks()` might already be reporting on deprecated interfaces. Assuming the interface is in use.

Reply:
Hmm, we don't have a way to lock an interface and this should ideally be pretty rare since we don't have Storm APIs for adding interfaces yet. This would only be hit if a user has added a form that uses the taxonomy interface and might make sense for it to be a bit noisy until 3.0.0?

Reply:
Agree on leaving it noisy until 3.0.